### PR TITLE
fix: qualify inputs on handling join in promql

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR:
- fix the bug that when aggr matcher is `None`, the planner forgets to update its tag column list
- add qualifier alias to join inputs to handle duplicate column name
- add corresponding tests

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by yuanbohan — APPROVED** [review]

**Comment by waynexia** [comment]
original error log:
image

**Review by evenyag — APPROVED** [review]
LGTM. To be honest, I gave up thinking about the prom planner.

File: src/promql/src/planner.rs [hunk 1]
```diff
@@ -261,24 +257,34 @@ impl PromPlanner {
...
+                        let left_table_ref: OwnedTableReference =
+                            self.ctx.table_name.clone().unwrap_or_default().into();
 
                         let right_input = self.prom_expr_to_plan(*rhs.clone()).await?;
                         let right_field_columns = self.ctx.field_columns.clone();
-                        let right_schema = right_input.schema().clone();
+                        let right_table_ref: OwnedTableReference =
+                            self.ctx.table_name.clone().unwrap_or_default().into();
+
+                        // TODO(ruihang): avoid join if left and right are the same table
```
Comment (killme2008) [inline src/promql/src/planner.rs:268]: I think we can return an error directly instead of leaving a TODO here.

Comment (waynexia) [reply src/promql/src/planner.rs:268]: No, it's supported now. This todo is an optimization

Comment (killme2008) [reply src/promql/src/planner.rs:268]: ok

**Pull request merged**